collectors/apps.plugin/apps_plugin.c

```diff
@@ -107,6 +108,25 @@ static int
 static char *user_config_dir = CONFIG_DIR;
 static char *stock_config_dir = LIBCONFIG_DIR;
 
+// some variables for keeping track of processes count by states
+typedef enum {
+    PROC_STATUS_RUNNING = 0,
+    PROC_STATUS_SLEEPING, // interruptible sleep
+    PROC_STATUS_SLEEPING_D, // uninterruptible sleep
+    PROC_STATUS_ZOMBIE,
+    PROC_STATUS_STOPPED,
+    PROC_STATUS_END,  //place holder for ending enum fields
+} proc_state;
+
+static proc_state proc_state_count[PROC_STATUS_END];
+static const char *proc_states[] = {
+    [PROC_STATUS_RUNNING] = " running",
+    [PROC_STATUS_SLEEPING] = " sleeping(interruptible)",
+    [PROC_STATUS_SLEEPING_D] = " sleeping(uninterruptible)",
+    [PROC_STATUS_ZOMBIE] = " zombie",
+    [PROC_STATUS_STOPPED] = " stopped",
```

Comment:
```suggestion
    [PROC_STATUS_RUNNING] = "running",
    [PROC_STATUS_SLEEPING] = "interruptible_sleep",
    [PROC_STATUS_SLEEPING_D] = "uninterruptible_sleep",
    [PROC_STATUS_ZOMBIE] = "zombie",
    [PROC_STATUS_STOPPED] = "stopped",
```

I think it is better to avoid `()` in dimension ids. cc @vlvkobal

Reply:
if we want (i personally think it is ok with `_`) `()` I suggest using [dimension name](https://learn.netdata.cloud/docs/agent/collectors/plugins.d#dimension)

> name
> the name of the dimension as it will appear at the legend of the chart, if empty or missing the id will be used

Reply:
since the process is in sleeping state, similar to running, it's better to use  'ing verb here. about parentheses, I agree

what about:
- sleeping_uninterruptible
- sleeping_interruptible

Reply:
Both look good to me (interruptible_sleep, sleeping_interruptible).